src/objects/player.gd

```diff
@@ -55,14 +50,34 @@ var previous_global_position: Vector3
 @onready var camera: Camera3D = %Camera3D
 @onready var footsteps: Node = %Footsteps
 @onready var shooting: Node = %Shooting
-@onready var punching = %Punching
+@onready var punching: Node = %Punching
+
+
+func _enter_tree() -> void:
+	var player_id := name.to_int()
+	set_multiplayer_authority(player_id)
 
 
 func _ready() -> void:
-	# We want finer control of the camera node, so it gets set as a top level node with interpolation disabled
-	camera.set_as_top_level(true)
-	normal_fov = camera.fov
-	# camera.set_physics_interpolation_mode(Node.PHYSICS_INTERPOLATION_MODE_OFF)
+	var player_id := name.to_int()
+	if player_id == multiplayer.get_unique_id():
+		# We want finer control of the camera node, so it gets set as a top level node with interpolation disabled
+		camera.set_as_top_level(true)
+		normal_fov = camera.fov
+		# camera.set_physics_interpolation_mode(Node.PHYSICS_INTERPOLATION_MODE_OFF)
```

Comment:
Did you mean to leave this comment in?

Reply:
Yes, Godot 3D physics interpolation isn't supported yet in Godot 4 but this line will be uncommented once it is.